### PR TITLE
[KVStore] Support delete_data() api on client

### PR DESCRIPTION
**Pull request by aksnzhy**
## Description

User can call client.delete_data('data_name') to delete tensor on kvstore.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/distributed/kvstore.py [hunk 1]
```diff
@@ -901,6 +950,37 @@ def init_data(self, name, shape, dtype, part_policy, init_func):
...
+        num_partitions = part_policy.partition_book.num_partitions()
+        num_clients_per_part = rpc.get_num_client() / num_partitions
+        if self._client_id % num_clients_per_part == 0:
+            request = DeleteDataRequest(name)
+            for n in range(self._group_count):
+                server_id = part_policy.part_id * self._group_count + n
+                rpc.send_request(server_id, request)
+            for _ in range(self._group_count):
+                response = rpc.recv_response()
+                assert response.msg == DELETE_MSG
```
Comment (zheng-da) [inline python/dgl/distributed/kvstore.py:974]: how do you delete the array in the backup servers?

Comment (aksnzhy) [reply python/dgl/distributed/kvstore.py:974]: The array in backup servers are deleted by server.

Comment (zheng-da) [reply python/dgl/distributed/kvstore.py:974]: i don't quite understand. the message is sent to every server or just the master server on each machine?

Comment (aksnzhy) [reply python/dgl/distributed/kvstore.py:974]: It will be sent to every server.

**Pull request merged**